### ISSUE TITLE
Recommend using concurrency instead of turnstyle

### ISSUE THREAD

**Issue by rhuanbarreto** [pulumi/ci-workflow-templates#13]
Some templates use the turnstyle action, which generates extra costs due to consuming time waiting for jobs to complete.

Instead they should use Github Actions concurrency feature: https://docs.github.com/en/actions/using-jobs/using-concurrency

**Comment by RobbieMcKinstry** [comment]
Good catch! There's no reason I see that we wouldn't prefer the `concurrency` field.
